conductr_cli/bndl_main.py

```diff
@@ -0,0 +1,597 @@
+from collections import OrderedDict
+from pyhocon import HOCONConverter, ConfigFactory, ConfigTree
+from conductr_cli import logging_setup
+from conductr_cli.constants import BNDL_PEEK_SIZE, IO_CHUNK_SIZE
+from conductr_cli.shazar_main import dir_to_zip, write_with_digest
+from io import BufferedReader, BytesIO
+import argcomplete
+import argparse
+import gzip
+import hashlib
+import json
+import logging
+import os
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+import zipfile
+
+mappings = OrderedDict([
+    ('--name', 'name'),
+    ('--version', 'version'),
+    ('--compatibility-version', 'compatibilityVersion'),
+    ('--system', 'system'),
+    ('--system-version', 'systemVersion'),
+    ('--nr-of-cpus', 'nrOfCpus'),
+    ('--memory', 'memory'),
+    ('--disk-space', 'diskSpace'),
+    ('--roles', 'roles')
+])
+
+
+def run(argv=None):
+    log = logging.getLogger(__name__)
+    parser = build_parser()
+    argcomplete.autocomplete(parser)
+    args = parser.parse_args(argv)
+
+    if args.source == '-':
+        args.source = None
+
+    if args.output == '-':
+        args.output = None
+
+    if sys.stdout.isatty() and sys.stdin.isatty() and args.source is None:
+        parser.print_help()
+    elif sys.stdout.isatty() and args.output is None:
+        log.error('bndl: Refusing to write to terminal. Provide -o or redirect elsewhere')
+        sys.exit(2)
+    else:
+        logging_setup.configure_logging(args)
+
+        sys.exit(args.func(args))
+
+
+def build_parser():
+    parser = argparse.ArgumentParser('bndl', formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument('-f', '--format',
+                        choices=['docker', 'oci-image'],
+                        required=False,
+                        help='The input format. When absent, auto-detection is attempted')
+
+    parser.add_argument('-t', '--tag',
+                        required=False,
+                        help='The name of the tag to use. For use with both docker and oci-image formats')
```

Comment:
How about this? `The name of the tag to create a ConductR bundle from. For use with docker and oci-image formats. When absent the first tag present is used.`